resolver.go

```diff
@@ -224,6 +224,22 @@ func createRespMsg(query *dns.Msg) *dns.Msg {
 	return resp
 }
 
+func (r *resolver) handleMXQuery(name string, query *dns.Msg) (*dns.Msg, error) {
+	addrv4, _ := r.backend.ResolveName(name, types.IPv4)
+	addrv6, _ := r.backend.ResolveName(name, types.IPv6)
+
+	if addrv4 == nil && addrv6 == nil {
+		return nil, nil
+	}
+
+	// We were able to resolve the name. Respond with an empty list with
```

Comment:
I see this comment but not sure how is it implemented below? does it mean that is enough to just reply with the same records?

Reply:
If we are able to resolve the hostname, for MX, we need to reply with a DNS msg with empty answers section (with status set to default RcodeSuccess). `createRespMsg` generates the DNS msg with empty answers section and status set to RcodeSuccess. This is very similar to what we are doing down below for ipv6 AAAA querries that resolve to a ipv4 address.